devito/types/dimension.py

```diff
@@ -536,6 +535,77 @@ def _arg_check(self, *args, **kwargs):
 # The Dimensions below are exposed in the user API. They can only be created by
 # the user
 
+class Thickness(DataSymbol):
```

Comment:
To be extra clear, but not required for this PR, we have two types of thicknesses:

1) those for classic SubDomains (which inherit from DataSymbol)
2) those that are "compiler generated" and therefore invisible to the user, like those created by an ABox (so we may need AbstractThickness, etc etc)

But again, for another day

Reply:
never mind, ignore the above... I see what you've done below now. Excellent